### PR TITLE
Renders Infinity as null

### LINKED ISSUE #221: 'Infinite' numbers are returned in JSON output
#  🐛 Bug Report

If the serialized data contains a number field with an `Infinite` value it's being serialized as `{"value": Infinite}` which produces an invalid JSON.

Is this an expected behavior?

## To Reproduce

```js
const fast = require("fast-json-stringify")
const stfy  = fast({
  type: 'object',
  required: ['a'],
  properties: {
    a: { type: 'number' }
  }
})
const obj = { a: Infinity }
console.log(stfy(obj)) // => "{\"a\":-Infinity}"
console.log(JSON.stringify(obj)) // => "{\"a\":null}"
```
Runkit: https://runkit.com/maximilian-krauss/5eb27a8a3651fb001a5b6521

## Expected behavior

Output should be `null`.

## Your Environment

- *node version*: 12
- *fastify version*: >=2.0.0
- *os*: Mac

### PR DESCRIPTION
**Pull request by maximilian-krauss**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

Closes #221

### CODE REVIEW

File: test/infinity.test.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+  const schema = {
+    title: 'integer',
+    type: 'integer'
+  }
+
+  const stringify = build(schema)
+  const output = stringify(Infinity)
+
+  t.equal(output, 'null')
```
Comment (Eomm) [inline test/infinity.test.js:19]: I think this should be `0`

For this schema:

```
{
  title: 'integer',
  type: 'integer'
}
```

the user wants only integer as output

For this one instead, user accepts `null` value also
```
{
  title: 'integer',
  nullable: true,
  type: 'integer'
}
```

Comment (maximilian-krauss) [reply test/infinity.test.js:19]: Sure, will do :)

Comment (jsumners) [reply test/infinity.test.js:19]: @Eomm I'm not sure it should be `0`. That has a very different meaning than `Infinity`. But I'm also not sure what to do here. The spec merely says:

https://tools.ietf.org/html/rfc8259#section-6
> Numeric values that cannot be represented in the grammar below (such
   as Infinity and NaN) are not permitted.

**Comment by maximilian-krauss** [comment]
Reverted PR back to draft, I added more tests and saw that nested objects are not rendered as expected. Will reopen when I'm done

File: test/infinity.test.js (hunk 1 above)
Comment (Eomm) [reply test/infinity.test.js:19]: You are right @jsumners 

I would just don't output the field (instead of zero) and output `null` when it is nullable.
I say so because when a JSON doesn't fit a schema, `fast-json-stringify` simply doesn't process it (like `foo: {type: 'string'}` and but then foo is an array)

Comment (maximilian-krauss) [reply test/infinity.test.js:19]: @Eomm  ok cool, then I will change it that way.
One question I have though, what's then the right output if I pass `{ type: 'integer' }` as schema and `Infinity` as value? `` or `undefined`?

Comment (Eomm) [reply test/infinity.test.js:19]: Edge cases 🎉 

The target is to serialize only a number (odd, but it is ok):

```js
schema = { type: 'integer' }
serialize = fjs(schema)
serialize(Infinity)
```

I would avoid undefined because this would break clients that do `JSON.parse(undefined)`
For this reason, I would say `''`

Do you agree @jsumners ?

Comment (jsumners) [reply test/infinity.test.js:19]: It should just be omitted or generate an error. The JSON Schema spec defers to the JSON spec — https://datatracker.ietf.org/doc/html/draft-handrews-json-schema-02#section-6.1

Therefore, `{ foo: Infinity }` is invalid.

Comment (Eomm) [reply test/infinity.test.js:19]: We do so when want to serialize a json, but when we try to serialize a number only?
It is doable for this module and I quite agree to throw a "CANNOT SERIALIZE" error.

It would be the first error thrown TBH

Comment (jsumners) [reply test/infinity.test.js:19]: It is clearly an error according to spec.

**Comment by mcollina** [comment]
Are you still interested in pursuing this?

**Comment by maximilian-krauss** [comment]
Hey @mcollina, oh I totally forgot about this one. Yes, I would give it another spin since we now know that it should be an error case :)
